css, main page, single search layout

ConnorMayhew opened the pull request.

ConnorMayhew committed:
CSS & main page udpates

updated css. Changed class names to better reflect UI diagrams. Some classes may yet be added in the future. Added more pages to main page, added communication from main page to single search page. Changed classes in main menu JS file to use renamed CSS classes.

ConnorMayhew committed:
Single Search Page Layout

Single Search page css layout, communication with main page. shows only function button relevant to function chosen on main page. main menu does not yet reflect this communication.  Adds single_search.js file which will handle generalizable JS functions

JojoVes commented on js/main_page.js:
```diff
@@ -0,0 +1,18 @@
+/*
```
Add something like [this](https://github.com/upeipls/UPLS/blob/4f6ba2191d46111c29551f71b8f9f2aeb47a19e4/js/edit_header.js#L1-L10) as well to enforce sign-in

JojoVes commented on js/single_search.js:
```diff
@@ -0,0 +1,46 @@
+/*
```
Add [this](https://github.com/upeipls/UPLS/blob/4f6ba2191d46111c29551f71b8f9f2aeb47a19e4/js/edit_header.js#L1-L10), same as previous comment.

JojoVes commented on main_page.html:
```diff
@@ -5,47 +5,105 @@
 		</div>
 	</center>
+	
+	<script>
```
All of these can just be switched to a more concise approach in the html for each button. Just add onclick="yourFunctionName()".

ConnorMayhew replied:
kept id-based approach, moved anonymous functions into js file (as discussed).

JojoVes commented on single_search.html:
```diff
@@ -2,12 +2,110 @@
+		</div>
+		
+		<script>
+		
```
Same thing as my previous comment of adding onclick="functionName()"

ConnorMayhew replied:
kept id-based approach, moved anonymous functions into js file (as discussed).

JojoVes commented on main_page.html:
```diff
@@ -5,47 +5,105 @@
 	<link rel="stylesheet" type="text/css" href="css/upls_styles.css">
-	<script src="js/main_menu.js"></script>
+	<script defer src="js/main_menu.js"></script>
+	<script defer src="js/main_page.js"></script>
```
Don't forget to include [these files](https://github.com/upeipls/UPLS/blob/4f6ba2191d46111c29551f71b8f9f2aeb47a19e4/group_email.html#L39-L40) too.

JojoVes reviewed (commented):
Just a few small things.

ConnorMayhew committed:
revision to main page & single search

revised based on comments from johanna

ConnorMayhew committed:
main page fix

messed up the js for login enforcement. Just changes some syntax. Also, removed aggregate button from main page.

ConnorMayhew committed:
single search field checking

checks fields for entry data and gives feedback to user

JojoVes commented on main_page.html:
```diff
@@ -6,46 +6,59 @@
 	<title>Main Page</title>
 	<link rel="stylesheet" type="text/css" href="css/upls_styles.css">
 	<script src="js/main_menu.js"></script>
+	<script src="js/main_page.js"></script>
```
js/main_page.js should be the last included file, because the api stuff needs to be included in order for signin stuff to work

JojoVes commented on single_search.html:
```diff
@@ -2,12 +2,85 @@
+		<link rel="stylesheet" type="text/css" href="css/upls_styles.css">
+	
+		<script src="js/main_menu.js"></script>
+		<script src="js/single_search.js"></script>
```
once again, just include js/sing_search.js last so api stuff can work

JojoVes reviewed (commented):
simple small copy/paste fixes to reorder some js includes

ConnorMayhew committed:
search functionality & minor fixes

adds search and display functionality to the single search page. Fixes external js inclusion order. Some small issues include the searches being case sensitive, further testing required to reveal the cause of some incorrect returns. Search functionality referenced from group email page

ConnorMayhew committed:
Merge remote-tracking branch 'upstream/master'

JojoVes commented on js/single_search.js:
```diff
@@ -0,0 +1,273 @@
+ 
+let sa = new SheetsApi();
+//!!set key!!-- remove this in final product (at least remove keys in pull requests)	
+var sheetID = "";
```
Remove this section.

ConnorMayhew committed:
Update single_search.js
